src/main/groovy/org/shipkit/internal/notes/contributors/AllContributorsFetcher.java

```diff
@@ -40,16 +43,36 @@ ProjectContributorsSet fetchAllContributorsForProject(String repository, String
         return result;
     }
 
-    private Set<ProjectContributor> extractContributors(List<JsonObject> page, String readOnlyAuthToken) throws IOException, DeserializationException {
+    private Set<ProjectContributor> extractContributors(List<JsonObject> page, final String readOnlyAuthToken) throws IOException, DeserializationException {
         Set<ProjectContributor> result = new HashSet<ProjectContributor>();
-        for (JsonObject contributor : page) {
-            //Since returned contributor does not have 'name' element, we need to fetch the user data to get his name
-            //TODO add static caching of this. Names don't change that often, let's just cache this forever in build cache.
-            String url = (String) contributor.get("url");
-            GitHubObjectFetcher userFetcher = new GitHubObjectFetcher(url, readOnlyAuthToken);
-            JsonObject user = userFetcher.getPage();
-            result.add(GitHubAllContributorsJson.toContributor(contributor, user));
+        //Since returned contributor does not have 'name' element, we need to fetch the user data to get his name
+        //TODO add static caching of this. Names don't change that often, let's just cache this forever in build cache.
+        ExecutorService executor = Executors.newFixedThreadPool(4);
+        int maxSizePerCallable = 25;
+
+        List<Future<Set<ProjectContributor>>> futures = new ArrayList<Future<Set<ProjectContributor>>>();
+        if (page.size() > 0) {
+            for (int i = 0; i < Math.max(page.size() / maxSizePerCallable, 1); i++) {
```

Comment:
I think we can come up with a simpler technical solution, with less math for paging.

High level guidelines:
 - decouple concurrency handling as much as possible from AllContributorsFetcher
 - decouple concurrency in a way it is unit testable without hitting GitHub

Suggested approach:
1. In AllContributorsFetcher 'page' object is a list of contributors 
2. We kick off threads, each thread: 
 - removes contributor from page
 - hits GitHub for more data for this contributor
 - converts data to the user
 - puts user into shared result list
3. we need to use locks or collections that are thread safe

Feel free to implement a solution as you see fit. Above is a suggestion :)